src/charm.py

```diff
@@ -411,11 +433,20 @@ def purge_k8s_resources(self, _: RelationDepartedEvent) -> None:
     def _on_config_changed(self, _: ConfigChangedEvent) -> None:
         """Handle configuration Change."""
         if self.update_stored_state("default-storage", "default_storage_class"):
-            logger.info(
-                "Config changed. New value: 'default-storage' = '%s'",
-                self.config.get("default-storage"),
-            )
-            self.update_storage_classes()
+            default_class = self.config.get("default-storage")
+            logger.info("Config changed. New value: 'default-storage' = '%s'", default_class)
+            try:
+                self.update_default_storage_class(default_class)
+            except ValueError:
+                self.unit.status = BlockedStatus(
+                    "Bad configuration option for 'default-storage'. See logs for more info.'"
+                )
+                return
+
+        # Resolve blocked state caused by bad configuration
+        status = self.unit.status
+        if status.name == BlockedStatus.name and status.message.startswith("Bad configuration"):
```

Comment:
nitpick: What do you say to use constant (global or in this function) with block status  `BlockedStatus("Bad configuration ...")`. Then check the `blocked_deault_storage_status` with `shelf.unit.status`. I think it will stand out more.

Reply:
I went with "prefix" and using `startswith` so that the same prefix can be reused by future config option to set blocked state and this single statement can clear any config-related blocked state.

Reply:
what would be the "storage" argument in this case? Blocked state is not meant to be tied to "storage", but to the fact that it was caused by configuration error.

My intention with this was that if the `_on_config_changed` method reaches bottom (meaning that there were no errors with configuration) it will resolve `Blocked` state of the unit, but only if the `Blocked` state was previously set by configuration error (hence the `Bad configuration` prefix).

Reply:
No problem at all, I appreciate thorough review and suggestions.